source/1_engine/rex_engine/include/rex_engine/string/stringid.h

```diff
@@ -0,0 +1,60 @@
+#pragma once
+
+#include "rex_engine/string/stringentry.h"
+#include "rex_engine/types.h"
+#include "rex_std/ostream.h"
+#include "rex_std/string.h"
+
+namespace rex
+{
+  class StringID
+  {
+  public:
+    StringID();
+    explicit StringID(rsl::string_view characters);
+
+    explicit operator u32() const;
+
+    bool operator==(const StringID& other) const;
+    bool operator!=(const StringID& other) const;
+
+    bool operator==(const StringEntryID& entryID) const;
+    bool operator!=(const StringEntryID& entryID) const;
+
+    rsl::string_view to_string_view() const;
+    bool is_none() const;
+    u32 value() const;
+
+  private:
+    /** Index into the StringID array (used to find String portion of the string/number pair used for comparison) */
+    StringEntryID m_comparison_index;
```

Comment:
this is not really an index but a key into the string pool which is a hash map.
because you used the term "index" I thought it was an array

Reply:
So you want me to rename it or is this just a general comment?

Reply:
please rename it

Reply:
also update the comment above it "Index into the StringID array"
it's not an array, it's hash table